NDTensors/src/dense/tensoralgebra/contract.jl

```diff
@@ -393,15 +400,34 @@ function _contract!(
 
   #tC = similar(CM)
   #_gemm!(tA, tB, El(α), AM, BM, El(β), CM)
-  @strided mul!(CM, AM, BM, El(α), El(β))
+  CM = mul!!(
+    leaf_parenttype(CM), CM, leaf_parenttype(AM), AM, leaf_parenttype(BM), BM, El(α), El(β)
+  )
```

Comment:
Same comment that I made above for `permutedims!!`, let's just use:
```julia
  CM = mul!!(CM, AM, BM, El(α), El(β))
```
and then define `mul!!` with a layer that then does the trait dispatch on `leaf_parenttype`.